src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -33,10 +40,20 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
                 request.getSession().setAttribute(ADMIN_MESSAGE, "You are not an admin!");
             }
         } else {
-            response.sendRedirect("/login");
+            request.setAttribute(ERROR, "You must log in before you can view the admin page!");
+            request.getRequestDispatcher("/WEB-INF/view/login.jsp").forward(request, response);
             return;
         }
 
+        int numUsers = UserStore.getInstance().getNumUsers();
```

Comment:
Here I think is where the lines could be moved to the admin.jsp file since the code is setting the admin stats, setting an attribute, and then that attribute is later called in the admin.jsp file. It seems a little redundant, but it might be better to do the processing in the servlet rather than in the .jsp file. 

Edit: I mean lines 48-50 for the lines that might be able to be put in the admin.jsp file

Reply:
Hi Sergio, thanks for the suggestion! Do you mean lines 46-48 or 50-52?

Reply:
Thanks Sergio! Just changed this. Could you review my recent changes and see if everything's okay? I've looked over your changes too (including the last refactoring commit to a helper method) and I think we can merge it in.

Reply:
Reminder so we don't make the same mistake I did: Rebase and squash the commits immediately before we merge!